In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

### **Question Number 1**

In [3]:
df=pd.read_csv('/content/Cutlets (1).csv')

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Unit A  35 non-null     float64
 1   Unit B  35 non-null     float64
dtypes: float64(2)
memory usage: 688.0 bytes


In [5]:
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

From the given data we have

Significance Level=5% = 0.05

Alpha=0.05

The Dataset of cutlets is having two units,so we are going to conduct 2 sample t test of two independent features .

For that first we need to check whether these features are following **Normal ditribution** or not.

We are going to use **shapiro** library for checking the normality of the features.

In [6]:
Alpha=0.05

**Normality Checking For Unit A**

In [7]:
from scipy.stats import shapiro

H0='The data is normal'
Ha='The data is not normal'


p=round(shapiro(df['Unit A'])[1],2)

if p>Alpha:
  print(f"{p}>{Alpha}.we fail to reject Null Hypothesis ",H0)
else:
  print(f"{p}<{Alpha}.we fail to accept Null Hypothesis ",Ha)

0.32>0.05.we fail to reject Null Hypothesis  The data is normal


**Normality Checking For Unit B**

In [8]:
from scipy.stats import shapiro

H0='The data is normal'
Ha='The data is not normal'


p=round(shapiro(df['Unit B'])[1],2)

if p>Alpha:
  print(f"{p}>{Alpha}.we fail to reject Null Hypothesis ",H0)
else:
  print(f"{p}<{Alpha}.we fail to accept Null Hypothesis ",Ha)

0.52>0.05.we fail to reject Null Hypothesis  The data is normal


So Both Features are **Having Normal Distributio**n

Testing Whether The Diameters Of Features are Same or Not

In [10]:
H0='Diameter of Unit A and Unit B are same '
Ha='Diameter of Unit A and Unib B are Different'

# we use ttest_ind for this problem
from scipy.stats import ttest_ind
P=round(ttest_ind(df['Unit A'],df['Unit B'])[1],2)

if P>Alpha:
  print(f'{P}>{Alpha} : We fail to reject Null Hypothesis ',H0)
else:
  print(f'{P}<{Alpha} : We fail to accept Null Hypothesis ',Ha)

0.47>0.05 : We fail to reject Null Hypothesis  Diameter of Unit A and Unit B are same 


**Final Output:** We fail to reject Null Hypothesis  Diameter of Unit A and Unit B are same

### Question Number 2

In [11]:
df2=pd.read_csv('/content/LabTAT.csv')

In [12]:
df2.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

We are going to Conduct ANOVA test on the given dataset ,becuase the it is not possible to check whether there is any difference in more than 2 groups ,so we have 4 group of continuous data for that ANOVA Hypothesis test is the best solution.

So we need to check whether there is difference in mean or not

**Normality Check**

In [13]:
from scipy.stats import shapiro

H0='The data is normal'
Ha='The data is not normal'


p=round(shapiro(df2['Laboratory 1'])[1],2)

if p>Alpha:
  print(f"{p}>{Alpha}.we fail to reject Null Hypothesis ",H0)
else:
  print(f"{p}<{Alpha}.we fail to accept Null Hypothesis ",Ha)

0.55>0.05.we fail to reject Null Hypothesis  The data is normal


In [14]:
from scipy.stats import shapiro

H0='The data is normal'
Ha='The data is not normal'


p=round(shapiro(df2['Laboratory 2'])[1],2)

if p>Alpha:
  print(f"{p}>{Alpha}.we fail to reject Null Hypothesis ",H0)
else:
  print(f"{p}<{Alpha}.we fail to accept Null Hypothesis ",Ha)

0.86>0.05.we fail to reject Null Hypothesis  The data is normal


In [15]:
from scipy.stats import shapiro

H0='The data is normal'
Ha='The data is not normal'


p=round(shapiro(df2['Laboratory 3'])[1],2)

if p>Alpha:
  print(f"{p}>{Alpha}.we fail to reject Null Hypothesis ",H0)
else:
  print(f"{p}<{Alpha}.we fail to accept Null Hypothesis ",Ha)

0.42>0.05.we fail to reject Null Hypothesis  The data is normal


In [16]:
from scipy.stats import shapiro

H0='The data is normal'
Ha='The data is not normal'


p=round(shapiro(df2['Laboratory 4'])[1],2)

if p>Alpha:
  print(f"{p}>{Alpha}.we fail to reject Null Hypothesis ",H0)
else:
  print(f"{p}<{Alpha}.we fail to accept Null Hypothesis ",Ha)

0.66>0.05.we fail to reject Null Hypothesis  The data is normal


So The All Four features are in Normal Distribution.

Now we need to go for f_test for calculating,whether there is difference in Groups or Not

In [26]:
pf_oneway(df2['Laboratory 1'],df2['Laboratory 2'],df2['Laboratory 3'],df2['Laboratory 4'])[1]

In [27]:
p

0.0

In [29]:
H0='There is No Difference in Groups'
Ha='There Exist Difference in Groups'

Alpha=0.05

from scipy.stats import f_oneway
p=f_oneway(df2['Laboratory 1'],df2['Laboratory 2'],df2['Laboratory 3'],df2['Laboratory 4'])[1]

if p>Alpha:
  print(f"{p}>{Alpha} : We fail to reject Null Hypothesis ",H0)
else:
  print(f"{p}<{Alpha} : We fail to accept Null Hypothesis ",Ha)

2.1156708949992414e-57<0.05 : We fail to accept Null Hypothesis  There Exist Difference in Groups


**Final Output**: 2.1156708949992414e-57<0.05 : We fail to accept Null Hypothesis  There Exist Difference in Groups

### **Question Number 3**

For the given problem we need to use Chi-Square test because the data in the dataset is categorical  for analyzing the categorical data we use **Chi_Square** Hypothesis test

In the problem said that if p values is less than Alpha regect Null Hypothesis

In [41]:
df3=pd.read_csv('/content/BuyerRatio.csv',index_col=0)

In [42]:
df3.head()

East  West  North  South
Observed Values                          
Males              50   142    131     70
Females           435  1523   1356    750

In [43]:
from scipy.stats import chi2_contingency

In [44]:
df3=chi2_contingency(df3)

In [46]:
df3

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [47]:
H0='All Proportions are Equal'
H1='Not All Proportions are Equal'

Alpha=0.05

p=round(df3[1],2)

if p>Alpha:
  print(f"{p}>{Alpha} : We Fail To reject Null Hypothesis ",H0)
else:
  print(f"{p}<{Alpha} : We Fail To Accept Null Hypothesis ",H1)

0.66>0.05 : We Fail To reject Null Hypothesis  All Proportions are Equal


**Final Output**:0.66>0.05 : We Fail To reject Null Hypothesis  All Proportions are Equal

### Question Number 4

We Are going to conduct test independece using Chi-Square hypothesis test with contingency table

In [53]:
df4=pd.read_csv('/content/Costomer+OrderForm.csv')

In [54]:
df4.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

Checking Value Counts in Data

In [59]:
print(df4['Phillippines'].value_counts(),'\n','\n',df4['Indonesia'].value_counts(),'\n','\n',df4['Malta'].value_counts(),'\n','\n',df4['India'].value_counts())

Error Free    271
Defective      29
Name: Phillippines, dtype: int64 
 
 Error Free    267
Defective      33
Name: Indonesia, dtype: int64 
 
 Error Free    269
Defective      31
Name: Malta, dtype: int64 
 
 Error Free    280
Defective      20
Name: India, dtype: int64


## Creating Contingency Table

In [60]:
contingency_table=[[271,267,269,280],[29,33,31,20]]
print(contingency_table)

[[271, 267, 269, 280], [29, 33, 31, 20]]


In [61]:
H0='The defective % varies by Center'
H1='The defective % not varies by Center'

Alpha=0.05

from scipy.stats import chi2_contingency

p=round(chi2_contingency(contingency_table)[1],2)

if p>Alpha:
  print(f"{p}>{Alpha} : We Fail To reject Null Hypothesis ",H0)
else:
  print(f"{p}<{Alpha} : We Fail To Accept Null Hypothesis ",H1)


0.28>0.05 : We Fail To reject Null Hypothesis  The defective % varies by Center


**Final Output**:0.28>0.05 : We Fail To reject Null Hypothesis  The defective % varies by Center